code/game/machinery/machinery.dm

```diff
@@ -125,6 +125,12 @@
 */
 /obj/machinery/proc/power_change()
 	var/old_stat = stat
+	if(machine_powernet.powernet_area != get_area(src))
+		var/area/machine_area = get_area(src)
+		if(machine_area)
+			machine_powernet.unregister_machine(src)
+			machine_powernet = machine_area.powernet
+			machine_powernet.register_machine(src)
```

Comment:
I recommend having the machines reregister to a powernet when they're reanchored / wrenched

Reply:
I've moved the registering of machines to the wrenching proc in machine.